src/vs/workbench/browser/positronModalDialogs/newFolderFromGitModalDialog.tsx

```diff
@@ -113,8 +113,10 @@ export const NewFolderFromGitModalDialog = (props: NewFolderFromGitModalDialogPr
 				if (isInputEmpty(result.repo)) {
 					throw new Error(localize('positron.gitRepoNotProvided', "A git repository URL was not provided."));
 				}
-				await props.createFolder(result);
+				// Dispose dialog immediately, then start cloning
+				const repoInfo = { ...result };
 				props.renderer.dispose();
+				await props.createFolder(repoInfo);
```

Comment:
I think we can pass `result` directly here!

```suggestion
				props.renderer.dispose();
				await props.createFolder(result);
```